### PR TITLE
Added IBDcompute + tests

### PR DESCRIPTION
**Pull request by magicDGS**
Not compatible with previous versions, because it is merging the IBD intervals correctly.

### CODE REVIEW

File: src/test/java/org/magicdgs/thaplv/tools/ibd/IBDcomputeIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class IBDcomputeIntegrationTest extends CommandLineProgramTest {
+
+    private final String expectedPrefix = "expected";
```
Comment (sputnikci) [inline src/test/java/org/magicdgs/thaplv/tools/ibd/IBDcomputeIntegrationTest.java]: \[FindBugs\] INFO: SS: Unread field: org\.magicdgs\.thaplv\.tools\.ibd\.IBDcomputeIntegrationTest\.expectedPrefix; should this field be static?

Comment (magicDGS) [reply src/test/java/org/magicdgs/thaplv/tools/ibd/IBDcomputeIntegrationTest.java]: Done

File: src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@CommandLineProgramProperties(
+        summary =
+                "Identify identity-by-descent (IBD) regions using the approach described in Langley et. al 2012, Genetics 192(2). "
+                        + "Pairwise differences between individuals are computed in windows, and called as IBD tracks if the "
+                        + "differences per site are smaller than a threshold.",
+        oneLineSummary = "Compute IBD regions using pair-wise differences between individuals.",
+        programGroup = AlphaProgramGroup.class)
```
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java]: Create new program group for this.

File: src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    @Argument(fullName = "output-differences", shortName = "out-diff", doc = "Output the divergence per window (.diff file) in addition to the IBD tracks")
+    public boolean outputDiff = false;
+
+    protected IBDOutput output;
+
+    private VCFHeader header;
+
+    @Override
+    public boolean requiresReference() {
```
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java:85]: This requires the reference just for counting the number of Ns. Nevertheless, users could avoid this counting and consider the whole window. Add a TODO and open a ticket for include this behaviour.

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDcollector.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+    /**
+     * New IBD collector for the samples in a VCF header
+     *
+     * @param header     the header where the samples are stored
+     * @param nCounter   the class for count the Ns in the reference
+     * @param windowSize the window size
+     * @param windowStep the window step
+     */
+    public IBDcollector(final VCFHeader header, final FastaNsCounter nCounter, final int windowSize,
```
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDcollector.java:78]: Add a ticket for include only some samples using the GATK command line parameters in the tools and modifying the collector depending on how this is implemented in their framework.

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/PairwiseDifferencesWindow.java [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+        Utils.nonNull(counter, "null counter");
+        if (IntervalUtils.intervalIsOnDictionaryContig(interval, counter.getDictionary())) {
+            this.interval = interval;
+            this.isLast = interval.getEnd() == counter.getChromosomeLength(interval.getContig());
+        } else {
+            this.interval = interval.expandWithinContig(0, counter.getDictionary());
+            this.isLast = true;
+        }
+        // TODO: this is -1 to keep consistency with the first implementation
+        // TODO: we should check at some point if this is real
```
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/engine/PairwiseDifferencesWindow.java:90]: Add a ticket for this.

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+        // retrieve the window
+        final SimpleInterval win = pairIBDmap.get(pair.getPairNames());
+        // see if the window is bigger than the threshold and have sites
+        if (differencesPerSite(pair, window) < minDiff && pair.getNumberOfSites() != 0) {
+            // if there is a window present for this pair
+            if (win != null) {
+                // if it is intersecting
+                // TODO: using overlaps() and meregeWithContiguous() breaks compatibility
+                // TODO: this was checked by converting to bed file and joining intervals
+                // TODO: now a test is covering this, but keeping this TODO for other tools that use this class
```
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java:141]: Add a comment about this expected incompatibility in #20.

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    private void flushWriters() {
+        ibdTracksWriter.flush();
+        if (pairwiseDiffWritter != null) {
+            pairwiseDiffWritter.flush();
+        }
+    }
+
+    public void close() {
+        logger.debug("Ending pending windows");
+        // TODO: should we break compatibility using dontCheck = true?
```
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java:210]: Add a ticket about this TODO

File: src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java (hunk 2 above)
Comment (magicDGS) [inline src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java]: Done

File: src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java (hunk 3 above)
Comment (magicDGS) [reply src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java:85]: Done #23

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java (hunk 6 above)
Comment (magicDGS) [reply src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java:141]: Done.

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java (hunk 7 above)
Comment (magicDGS) [reply src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDOutput.java:210]: Done (#24)

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDcollector.java (hunk 4 above)
Comment (magicDGS) [reply src/main/java/org/magicdgs/thaplv/tools/ibd/engine/IBDcollector.java:78]: Done (#25)

File: src/main/java/org/magicdgs/thaplv/tools/ibd/engine/PairwiseDifferencesWindow.java (hunk 5 above)
Comment (magicDGS) [reply src/main/java/org/magicdgs/thaplv/tools/ibd/engine/PairwiseDifferencesWindow.java:90]: Done (#26)

**Pull request merged**